docs/conditional-gatherer/README.md

```diff
@@ -1,185 +1,137 @@
-`expr` value to `1 > bool 0` and `for` to `1s`:
-```bash
-echo '{
-    "apiVersion": "monitoring.coreos.com/v1",
-    "kind": "PrometheusRule",
-    "metadata": {
-        "name": "samples-operator-alerts",
-        "namespace": "openshift-cluster-samples-operator"
-    },
-    "spec": {
-        "groups": [
-            {
-                "name": "SamplesOperator",
-                "rules": [
-                    {
-                        "alert": "SamplesImagestreamImportFailing",
-                        "annotations": {
-                            "message": "Always firing"
-                        },
-                        "expr": "1 > bool 0",
-                        "for": "1s",
-                        "labels": {
-                            "severity": "warning"
-                        }
-                    }
-                ]
-            }
-        ]
-    }
-}' | oc apply -f -
-```
+### Pod Logs
 
-4. Wait for the alert to fire:
-```bash
-export PROMETHEUS_HOST=$(oc get route -n openshift-monitoring prometheus-k8s -o jsonpath='{@.spec.host}')
-export INSECURE_PROMETHEUS_TOKEN=$(oc get secret $(oc get sa prometheus-k8s -n openshift-monitoring -o json | jq .secrets[0].name | tr --delete \") -n openshift-monitoring -o json | jq .metadata.annotations.\"openshift.io/token-secret.value\" | tr --delete \")
-curl -g -s -k -H 'Cache-Control: no-cache' -H "Authorization: Bearer $INSECURE_PROMETHEUS_TOKEN" "https://$PROMETHEUS_HOST/api/v1/query" --data-urlencode 'query=ALERTS{alertstate="firing",alertname="SamplesImagestreamImportFailing"}' | jq ".data.result[]"
-```
+The `pod_logs` gathering function is a powerful tool to collect logs from specific pods based on various criteria. Here's an example configuration:
 
-The output should be:
 ```json
 {
-  "metric": {
-    "__name__": "ALERTS",
-    "alertname": "SamplesImagestreamImportFailing",
-    "alertstate": "firing",
-    "severity": "critical"
-  },
-  "value": [
-    1652363876.855,
-    "1"
-  ]
+    "pod_logs": {
+        "resource_filter": {
+            "namespaces": ["openshift-monitoring"],
+            "pod_name_regex_filter": "thanos-"
+        },
+        "log_filter": {
+            "messages_to_search": ["connect: connection refused"],
+            "tail_lines": 100,
+            "previous": false
+        }
+    }
 }
 ```
 
-5. Make metrics work by forwarding the endpoint and setting INSECURE_PROMETHEUS_TOKEN environment variable:
-```bash
-export INSECURE_PROMETHEUS_TOKEN=$(oc get secret $(oc get sa prometheus-k8s -n openshift-monitoring -o json | jq .secrets[0].name | tr --delete \") -n openshift-monitoring -o json | jq .metadata.annotations.\"openshift.io/token-secret.value\")
-```
-```bash
-# run this command in a separate terminal
-sudo kubefwd svc -n openshift-monitoring -d openshift-monitoring.svc -l app.kubernetes.io/instance=k8s --kubeconfig $KUBECONFIG
-```
-
-6. Run the operator and wait for an archive containing `conditional/` directory.
+In this example:
```

Comment:
Examples are useful, but I am rather looking for a reference guide that provides a comprehensive documentation of each gathering function and all its parameters.

Reply:
Nvm, a complete list is in one of the next sections.